### ISSUE TITLE
Enabling addons kills Friendica

### ISSUE THREAD

**Issue by jakedfw** [friendica/friendica#12811]
Hi all, when I go to enable an addon, it immediately blocks all pages, and they become white. I waited to see if it was server processing, but the white page never goes away. 

The second I disable the addon using bin/console the site comes back. 

This has occured for three different addons so it seems to be universal, not addon specific.

**Comment by jakedfw** [comment]
One odd thing I've noticed is that I have two directories:

addon
friendica-addons

They seem identical. Maybe a github install issue?

**Comment by jakedfw** [comment]
Just realized that the guide I used to install Friendica may have uploaded a dev version:
Friendica 'Giant Rhubarb' 2023.03-dev - 1514

Is there a way to make sure I stay on the stable branch at this point?

(Not sure if related)

**Comment by tobiasd** [comment]
To change from the development branch to the stable branch, run the command `git checkout stable` as long as you do not checkout develop, you should stay on stable then.

**Comment by jakedfw** [comment]
I'm sorry, can you dumb that down for me? :( 

I ran that in my server friendica directory and it said my directory was a 'dubious' repository.

**Comment by MrPetovan** [comment]
Can you please paste the complete output of the git command you ran?

**Comment by jakedfw** [comment]
<img width="1131" alt="image" src="https://user-images.githubusercontent.com/38843638/219965764-66a7ce06-c8de-4108-ac59-8e75a64ef8b0.png">

**Comment by jakedfw** [comment]
By the way I did try `git config --global --add safe.directory /var/www/friendica`

I got an `unknown option` error on --config

**Comment by MrPetovan** [comment]
The ownership issue arises from the fact you are running these commands as root when you should be running them as `www-data`.

Please run (as root) `chown -R www-data: /var/www/friendica` and then switch to the `www-data` user with `su - www-data` to perform any git or Friendica console commands.

In this state, you can go back to the root user at any time with `exit`.

**Comment by jakedfw** [comment]
Thank you! You are the best. Unfortunately, it was then aborted:

`www-data@Kerr-Social-New:~/friendica$ git checkout stable
error: Your local changes to the following files would be overwritten by checkout:
        CHANGELOG
        VERSION
        bin/testargs.php
        composer.lock
        database.sql
        doc/Addons.md
        doc/database.md
        doc/database/db_delivery-queue.md
        doc/database/db_
        ......`

**Comment by MrPetovan** [comment]
Ok, we can deal with that. You must first restore your project files to match the current branch:
```
git checkout -- .
```
Then you can safely switch branch to stable:
```
git checkout stable
```
You will need to perform both these commands in the `addon` folder as well.

**Comment by jakedfw** [comment]
Thank you. It says it can't update to stable as it will overwrite my files, but I assume that's expected?
(addon was already on stable it looks like)

**Comment by MrPetovan** [comment]
Ah, you also have to clean up new files that weren't versioned in your branch. The full command is the following:
```
git checkout -- . && git clean -df
```
Then you can check out the `stable` branch.

**Comment by jakedfw** [comment]
That did it! You are SO helpful. Thanks so much for dealing with this total amateur.

I'm guessing that I'll be updated to the next stable release and after that I'll version with the stable releases.

Thanks again!!

**Comment by MrPetovan** [comment]
> That did it! You are SO helpful. Thanks so much for dealing with this total amateur.

We all had to start somewhere, I try to be mindful of that when we receive support requests.

**Comment by jakedfw** [comment]
Oh no, I spoke too soon. I'm getting http error 500 on every page now. 

Any ideas?

**Comment by jakedfw** [comment]
(Just to be insanely careful, I even rebooted the server)

**Comment by MrPetovan** [comment]
Now it's time for the next step in your learning journey: locating logs. A 500 error will always leave a trace, either in your web server logs or your phpfpm logs.  Their location varies depending on the underlying OS, but a good place to start looking is `/var/log/`.

If this is fruitless, the next place to start looking is your web server/PHP configuration files. Again, it depends on architecture but a common place for these files who may reveal the location of the log file is `/etc`.

**Comment by jakedfw** [comment]
Here's the log from Apache (there were no Friendica site-specific errors I could find in the domain error log. They were all from yesterday and earlier):

[Mon Feb 20 00:00:01.210347 2023] [core:notice] [pid 17168] AH00094: Command line: '/usr/sbin/apache2'
sh: 1: /usr/sbin/sendmail: not found
sh: 1: /usr/sbin/sendmail: not found
[Mon Feb 20 17:47:30.757688 2023] [mpm_prefork:notice] [pid 17168] AH00170: caught SIGWINCH, shutting down gracefully
[Mon Feb 20 17:48:05.271678 2023] [mpm_prefork:notice] [pid 853] AH00163: Apache/2.4.55 (Ubuntu) OpenSSL/3.0.2 configured -- resuming normal ope>
[Mon Feb 20 17:48:05.307577 2023] [core:notice] [pid 853] AH00094: Command line: '/usr/sbin/apache2'

I don't have sendmail installed on purpose, but that's not a Friendica dependency that I can tell. 

Here's the phpfpm log:

  GNU nano 6.2                                                      php8.2-fpm.log                                                               
[19-Feb-2023 00:00:03] NOTICE: error log file re-opened
[20-Feb-2023 17:47:30] NOTICE: Terminating ...
[20-Feb-2023 17:47:30] NOTICE: exiting, bye-bye!
[20-Feb-2023 17:48:04] NOTICE: fpm is running, pid 722
[20-Feb-2023 17:48:04] NOTICE: ready to handle connections
[20-Feb-2023 17:48:04] NOTICE: systemd monitor interval set to 10000ms

**Comment by jakedfw** [comment]
Would it be possible for me to do a re-install of the files and retain the database? Looking to solve this and am okay with some desperate measures. :)

**Comment by MrPetovan** [comment]
You just need to keep the `config/local.config.php` file if you go the nuclear route.

**Comment by jakedfw** [comment]
I'm going nuclear! Wish me luck!

**Comment by jakedfw** [comment]
www-data@Kerr-Social-New:~/friendica$ bin/console autoinstall -v
Executable: /var/www/friendica/bin/console.php
Arguments: array (
  0 => 'autoinstall',
)
Options: array (
  'v' => 1,
)
Command: autoinstall
Initializing setup...
[Error] Friendica\Core\Session\Factory\Session::createSession(): Return value must be of type Friendica\Core\Session\Capability\IHandleSessions, null returned
[Backtrace]:
#0 /var/www/friendica/vendor/level-2/dice/Dice.php(150): Friendica\Core\Session\Factory\Session->createSession()
#1 /var/www/friendica/vendor/level-2/dice/Dice.php(96): Dice\Dice->Dice\{closure}()
#2 /var/www/friendica/vendor/level-2/dice/Dice.php(247): Dice\Dice->create()
#3 /var/www/friendica/vendor/level-2/dice/Dice.php(132): Dice\Dice->Dice\{closure}()
#4 /var/www/friendica/vendor/level-2/dice/Dice.php(96): Dice\Dice->Dice\{closure}()
#5 /var/www/friendica/src/DI.php(205): Dice\Dice->create()
#6 /var/www/friendica/src/Core/Installer.php(273): Friendica\DI::l10n()
#7 /var/www/friendica/src/Core/Installer.php(73): Friendica\Core\Installer->checkPHP()
#8 /var/www/friendica/src/Core/Installer.php(693): Friendica\Core\Installer->getPHPPath()
#9 /var/www/friendica/src/Console/AutomaticInstallation.php(123): Friendica\Core\Installer->setUpCache()
#10 /var/www/friendica/vendor/asika/simple-console/src/Console.php(108): Friendica\Console\AutomaticInstallation->doExecute()
#11 /var/www/friendica/src/Core/Console.php(163): Asika\SimpleConsole\Console->execute()
#12 /var/www/friendica/vendor/asika/simple-console/src/Console.php(108): Friendica\Core\Console->doExecute()
#13 /var/www/friendica/bin/console.php(41): Asika\SimpleConsole\Console->execute()
#14 {main}

**Comment by jakedfw** [comment]
Looks like it was the lack of .htaccess. I fixed that. But now I'm getting a `HTTP error 500`

A placeholder index.html works, so this doesn't seem to be an apache issue. SSL works, too. To summarize:

Copied config file to /tmp/
Deleted the folder with the developer build.
Ran the git clone of the stable build
did rest of install, including  chown's. 
Moved config file to config folder
ran install script
Boom 500 error

**Comment by jakedfw** [comment]
Hmm. I deleted and re-added the folder. Would that mess up Apache sites-available? The directory name and structure hasn't changed.

**Comment by MrPetovan** [comment]
What does your `config/local.config.php` file look mike (after removing the passwords)?

**Comment by jakedfw** [comment]
```
  GNU nano 6.2                                                     local.config.php                                                              
 *
 * If you're unsure about what any of the config keys below do, please check the static/defaults.config.php file for
 * detailed documentation of their data type and behavior.
 */

return [
        'database' => [
                'hostname' => 'localhost',
                'username' => 'friendica',
                'password' => '[password]',
                'database' => 'Friendica_db',
                'charset' => 'utf8mb4',
        ],

        // ****************************************************************
        // The configuration below will be overruled by the admin panel.
        // Changes made below will only have an effect if the database does
        // not contain any configuration for the friendica system.
        // ****************************************************************

        'config' => [
                'admin_email' => 'jake@jakekerr.com',
                'sitename' => 'The social web home of the Kerr family',
                'register_policy' => \Friendica\Module\Register::OPEN,
                'register_text' => '',
                'hostname' => 'friendica.local',
        ],
        'system' => [
                'default_timezone' => 'America/Chicago',
                'language' => 'en',
                'url' => 'https://kerr.social',
        ],
];
```

**Comment by jakedfw** [comment]
Was literally a copy and paste from the old install that worked.

**Comment by MrPetovan** [comment]
Ok, then you don't need to run the autoinstall script at all, your node should be able to work out of the box with both the config file and a copy of the stable release of Friendica. Let me know if it still isn't the case.

**Comment by jakedfw** [comment]
Should I re-install without the install script? Because I'm getting 500 errors.

**Comment by MrPetovan** [comment]
Sure, it can't hurt at this point. If it keeps erring out we'll have to try something else, but at the moment I feel like the autoinstall script is doing more harm than good.

And as always, please make sure your base install and you addon folder are both on the same release/git branch.

**Comment by jakedfw** [comment]
Did the re-install. 
Checked the webpage after every step. Site unreachable was fixed with the copy of .htaccess
Next up the web page came up saying to run bin/composer.phar --no-dev install
This is actually different than the install instructions so I used those. 
In the friendica directory, I ran bin/composer.phar --no-dev install 
That worked fine.
Click back to the web page
HTTP 500 error 
😭

**Comment by jakedfw** [comment]
Checked Apache2 error logs:
Access is full of Mastodon connections, which I assume are the workers populating the database. That seems working.
Error log for the site is empty.
PHP8.2 error log is also empty.

**Comment by jakedfw** [comment]
git branch --show-current
Shows `stable` for both main directory and addon

**Comment by jakedfw** [comment]
So I tried a fresh install, only this time I created a new database and used the graphical install. The graphical install worked until I got to the database selection. When I chose my older database I got this error:
<img width="791" alt="image" src="https://user-images.githubusercontent.com/38843638/220823309-a9b8f5e2-4fc8-4a2c-84a5-50223bceac8c.png">
When I installed with the new database it worked fine. So it definitely appears that you can't use an existing database for a new install (or at least you can't do it with a database created under development and then moved to stable).
